driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -598,8 +560,19 @@ public void onSet(Connection connection, Message.Response response, long latency
                                 return;
                             default:
                                 connection.release();
-                                if (metricsEnabled())
+                                if (err.code == SERVER_ERROR)
```

Comment:
You will notice that I removed the logger call, because it would be too verbose to log all kinds of errors, and because ultimately I think it's up to the retry policy to log the error if it needs to.